src/Microsoft.SqlTools.ServiceLayer/DacFx/DacFxService.cs

```diff
@@ -417,5 +432,46 @@ internal void PerformOperation(DacFxOperation operation, TaskExecutionMode taskE
         {
             operation.Execute(taskExecutionMode);
         }
+
+        /// <summary>
+        /// Changes telemetry status 
+        /// </summary>
+        private void UpdateTelemetryStatus(bool telemetryEnabled)
+        {
+            try
+            {
+                if (telemetryEnabled)
+                {
+                    DacServices.EnableTelemetry(telemetryApplicationName, serviceVersion);
+                }
+                else
+                {
+                    DacServices.DisableTelemetry();
+                }
+            
+            }
+            catch (Exception ex)
+            {
+                Logger.Warning($"Failed to update DacFx telemetry status. telemetry enable: {telemetryEnabled}, error: {ex.Message}");
+            }
+        }
+
+        private static Version LoadServiceVersion()
```

Comment:
Please make sure to use proper nullable types here - this should be `Version?` for example

I would suggest removing the `#nullable disable` command from this file and seeing how many there are to fix - if it's just a couple then it'd be nice to fix them now. Otherwise you can at least use that to easily see if the new code you added is using proper nullable types.

Reply:
I made the change but I get warning now because of "#nullable disable". should I still keep it?

Reply:
No, and in fact it would be great for everyone working in STS to be slowly trying to remove these by fixing up the nullable stuff bit by bit. So if you've done that for this file then you can remove the pragma and we're one step closer to having that complete!

Reply:
I enabled it back and fix the only nullable issue I saw. @Benjin @kisantia please let me know if you have any concern enabling nullable in this file